server/channels/api4/post.go

```diff
@@ -628,18 +628,31 @@ func deletePost(c *Context, w http.ResponseWriter, _ *http.Request) {
 		return
 	}
 
+	permanent := c.Params.Permanent
+
 	auditRec := c.MakeAuditRecord("deletePost", audit.Fail)
 	defer c.LogAuditRecWithLevel(auditRec, app.LevelContent)
 	audit.AddEventParameter(auditRec, "post_id", c.Params.PostId)
+	audit.AddEventParameter(auditRec, "permanent", permanent)
 
-	post, err := c.App.GetSinglePost(c.AppContext, c.Params.PostId, false)
+	includeDeleted := false
+	if permanent {
+		includeDeleted = true
+	}
+
+	post, err := c.App.GetSinglePost(c.AppContext, c.Params.PostId, includeDeleted)
 	if err != nil {
 		c.SetPermissionError(model.PermissionDeletePost)
 		return
 	}
 	auditRec.AddEventPriorState(post)
 	auditRec.AddEventObjectType("post")
 
+	if permanent && !c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionManageSystem) {
+		c.SetPermissionError(model.PermissionManageSystem)
+		return
+	}
```

Comment:
Do we need to be that restrictive? I expect that users can delete their own posts by default. That would be very helpful to have for GDPR compliance on public servers.

Should we introduce a new permission?

Reply:
I put it at the same level as the other permanent deletion tasks because the ask was for sysadmins to do this and for other compliance reasons I think it's better to just do it this way for now. Regular users can use the soft delete and if they want a GDPR deletion they can talk to the sysadmin

Reply:
Fair point. I hoped to squeeze in some GDPR-related changes into the PR, but consistency is important.